applications/IgaApplication/custom_elements/shell_5p_element.cpp

```diff
@@ -501,7 +483,11 @@ namespace Kratos
         { return nodalVar * (node.*funct)(args...); };
         BoundedVector<double, 3> nullVec;
         nullVec = ZeroVector(3);
-        return std::inner_product(vec.begin(), vec.end(), GetGeometry().begin(), nullVec, std::plus<BoundedVector<double, 3>>(), nodeValuesTimesAnsatzFunction);
+        for (IndexType i = 0; i < vec.size(); ++i) {
+            nullVec += nodeValuesTimesAnsatzFunction(vec[i], GetGeometry()[i]);
```

Comment:
Ok this is also an acceptable solution. A **nice** range-based for loop is not possible unit C++20. Therefore we can keep it like that here.

However, the issue why `std::inner_product` didn't work would still be interesting.

Reply:
not sure if this helps either as we have both vectors - and one is the weird one. - probably the controversy of the pointer vector with access to pointer and vector might be the issue in the inner prod. But that is only a vague speculation...